bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/api/EcobeeApi.java

```diff
@@ -337,25 +338,34 @@ private boolean isSuccess(@Nullable AbstractResponseDTO response) {
         if (response == null) {
             logger.debug("API: Ecobee API returned null response");
         } else if (response.status.code.intValue() != 0) {
-            logger.info("API: Ecobee API returned unsuccessful status: code={}, message={}", response.status.code,
+            logger.debug("API: Ecobee API returned unsuccessful status: code={}, message={}", response.status.code,
                     response.status.message);
             if (response.status.code == ECOBEE_DEAUTHORIZED_TOKEN) {
                 // Token has been deauthorized, so restart the authorization process from the beginning
                 logger.warn("API: Reset OAuth Client Service due to deauthorized token");
                 deleteOAuthClientService();
                 createOAuthClientService();
             } else if (response.status.code == ECOBEE_TOKEN_EXPIRED) {
-                // Check isAuthorized again to see if we can get a valid token
-                logger.info("API: Unable to complete API call because token is expired. Try to refresh the token...");
+                logger.debug("API: Unable to complete API call because token is expired. Try to refresh the token...");
+                // Log some additional debug information about the current AccessTokenResponse
+                AccessTokenResponse localAccessTokenResponse = accessTokenResponse;
+                if (localAccessTokenResponse != null) {
+                    logger.debug("API: AccessTokenResponse created on: {}", localAccessTokenResponse.getCreatedOn());
+                    logger.debug("API: AccessTokenResponse expires in: {}", localAccessTokenResponse.getExpiresIn());
+                }
+                // Recreating the OAuthClientService seems to be the only way to handle this error
                 closeOAuthClientService();
                 createOAuthClientService();
-                if (isAuthorized()) {
-                    return true;
-                } else {
-                    logger.info("API: isAuthorized was NOT successful on second try");
+                try {
+                    oAuthClientService.refreshToken();
+                } catch (OAuthException | IOException | OAuthResponseException e) {
+                    logger.warn("API: Unable to refresh token after receiving error code 14");
+                    bridgeHandler.updateBridgeStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "Unable to refresh access token");
```

Comment:
```suggestion
                            "Unable to refresh access token, code 14");
```

If add sdomehting like this to the thing state, the logger can be removed as it does not add anything to the logging that is allready there.

Reply:
@lsiepel I'm a little confused. If you're logging debug to openhab.log and logging thing ONLINE/OFFLINE status changes to events.log, isn't it a hassle to be looking across 2 log files to get the full picture of what's going on? I must be missing something.

Reply:
It is a general rule to only log an issue once. Last bit of: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
I understand it's a general rule. And I agree there are many, many cases where it doesn't make sense, such as the rather obvious example given in the docs. I just don't understand why you wouldn't want to se an error condition in the log file, especially when the error condition in this context has been shown to occur only every few months. And in this scenario where there's other relevant information being logged to debug.

Reply:
I wasn't aware of this low frequency.

Seen some log viewers that combine both logs. If you manually check the logs seperate I can see your use case. Not going to maker an issue of it.

Reply:
Thanks. Yeah, a coule things have to go wrong here. First, you get the error code 14 from Ecobee which happens very rarely (every few months), then you get an exception trying to refresh the access token after having just successfully created the OAuthClientService.